src/gmail.js

```diff
@@ -2368,7 +2375,15 @@ var Gmail = function(localJQuery) {
                         if(!api.tracker.recipient_matches.length) return;
 
                         // determine an array of all emails specified for To, CC and BCC and extract addresses into an object for the callback
-                        var compose = new api.dom.compose(api.tracker.recipient_matches[0].closest("div.M9"));
+                        let composeRoot = api.tracker.recipient_matches[0].closest("div.M9");
+                        // sometimes (on copy-paste of contact in peoplekit mode) element disappears
+                        if (composeRoot.length === 0 && api.tracker.recipient_matches.length > 0) {
+                            composeRoot = api.tracker.recipient_matches[1].closest("div.M9");
+                        }
+                        if (composeRoot.length === 0) {
+                            api.tools.error("Can't find composeRoot for " + match)
```

Comment:
Keep in mind this code is running in the background while intercepting genuine Gmail API traffic. We do NOT want the presence of Gmail.JS to crash Gmail under *any* circumstances.

Having the Gmail.JS API fail is OK. Having the extension depending on Gmail.JS fail is OK. Having Gmail itself fail is *never* OK.

As such, I'd rather have this statement replaced by a generic `console.error()` statement, rather than calling into something which may or may not throw :smile:

Reply:
This was improved, please check. It should not affect GMail as it just does nothing if composeRoot not found

Reply:
Looking at this again, this seems not to be related to XHR interception at all (my bad), and in that case this code is perfectly acceptable. And the updated code is even better.

I'm OK with this as things are now :+1: